features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/Comments/Comment.js

```diff
@@ -1,103 +1,286 @@
-import { withStyles } from "@material-ui/core/styles";
-import Collapse from "@material-ui/core/Collapse";
-import DeleteOutlined from "@material-ui/icons/DeleteOutlined";
-import { Typography } from "@material-ui/core";
-import { ApiContext } from "../ApiContext";
-import AccountBox from "@material-ui/icons/AccountBox";
-import Paper from "@material-ui/core/Paper";
-import Grid from "@material-ui/core/Grid";
-import VerticalDivider from "../../../Shared/VerticalDivider";
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Typography } from '@material-ui/core/';
+import { withStyles } from '@material-ui/core/styles';
+import AccountBox from '@material-ui/icons/AccountBox';
+import Grid from '@material-ui/core/Grid';
+import Alert from '../../../Shared/Alert';
+import ConfirmDialog from '../../../Shared/ConfirmDialog';
 import CommentAdd from './CommentAdd';
+import CommentEdit from './CommentEdit';
+import CommentOptions from './CommentOptions';
+import CommentReply from './CommentReply';
+import API from '../../../../data/api';
 
 const styles = theme => ({
-  link: {
-    color: theme.palette.getContrastText(theme.palette.background.default),
-    cursor: "pointer"
-  },
-  time: {
-    color: theme.palette.getContrastText(theme.palette.background.default)
-  },
-  commentIcon: {
-    color: theme.palette.getContrastText(theme.palette.background.default)
-  },
-  commentText: {
-    color: theme.palette.getContrastText(theme.palette.background.default)
-  },
-  deleteIcon: {
-    cursor: 'pointer',
-    color: theme.palette.getContrastText(theme.palette.background.default),
-    width: 20,
-  },
-  root: {
-    marginTop: 20,
-  }
+    link: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+        cursor: 'pointer',
+    },
+    time: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+        marginTop: 3,
+    },
+    commentIcon: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+    },
+    commentText: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+        marginTop: 2,
+        width: '100%',
+        whiteSpace: 'pre-wrap',
+        overflowWrap: 'break-word',
+    },
+    root: {
+        marginTop: 20,
```

Comment:
Better to get the height from the theme space unit. ex marginTop: theme.spacing.unit*3